### PR TITLE
feat: Add support to configure the ESXi password expiration policy

### LINKED ISSUE #34: Add support to configure the ESXi password expiration policy
## Description

Add the ability to configure the ESXi Password Expiration Policy

## Use Case(s)

I'm a SRE (Site Resiliency Engineer) and need to be able to adjust the password expiration policy on multiple ESXi within my VMware Cloud Foundation instance based on a vSphere cluster.

**Community Note**

Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the maintainers prioritize this request.

Please do not leave "+1" or "me too" comments. These generate noise for issue followers and do not help prioritize the request.

If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### PR DESCRIPTION
**Pull request by garlicNova**
In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware-samples/power-validated-solutions-for-cloud-foundation/blob/main/CONTRIBUTING.md) for making a pull request.

**Summary of Pull Request**

Add cmdlet
- `Set-EsxiPasswordExpirationPeriod`

This cmdlet is use to set the password expiration period on an ESXi host in number of days.

Sign-off-by: Kevin Teng <kteng@vmware.com>

**Type of Pull Request**

- [ ] This is a bug fix.
- [x] This is an enhancement or feature.
- [ ] This is a code style / formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

Closes #34 

**Test and Documentation Coverage**

- [x] Tests have been completed (for bug fixes / features).
- [ ] Documentation has been added / updated (for bug fixes / features).

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

**Review by GaryJBlake — CHANGES_REQUESTED** [review]

File: PowerValidatedSolutions.psm1 [hunk 1]
```diff
@@ -491,6 +491,81 @@ Function Set-vCenterPasswordExpiration {
...
+		The cmdlet connects to SDDC Manager using the -server, -user, and -password values:
+        - Validates that network connectivity and authentication is possible to SDDC Manager
+        - Validates that the workload domain exists in the SDDC Manager inventory
+        - Validates that network connectivity and authentication is possible to vCenter Server
+        - Gathers the ESXi hosts for the cluster specificed
+        - Configured all ESXi hosts in he provided cluster
+
+        EXAMPLE
+        Set-EsxiPasswordExpirationPolicy -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -cluster sfo-m01-cl01 -ExpirationInDays 60
+        This example configures all ESXi hosts within the cluster named sfo-m01-cl01 of the workload domain sfo-m01
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:510]: Change to: "for the workload domain".

File: PowerValidatedSolutions.psm1 [hunk 2]
```diff
@@ -491,6 +491,81 @@ Function Set-vCenterPasswordExpiration {
...
+                                                if ($detail -eq "true") {
+                                                    Write-Output "Updating Advanced System Setting (Security.PasswordMaxDays) on ESXi Host ($esxiHost): SUCCESSFUL"
+                                                }
+                                            }
+                                            else {
+                                                Write-Error "Updating Advanced System Setting (Security.PasswordMaxDays) on ESXi Host ($esxiHost): POST_VALIDATION_FAILED"
+                                            }
+										}
+										else {
+											Write-Error "Error applying password policy"
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:546]: Does this need a `POST_VALIDATION_FAILED` msg?

**Review by tenthirtyam — CHANGES_REQUESTED** [review]
@garlicNova, can you make these minor changes and also ensure you update your git config with your `user.name` and `user.email` before the commit?

```bash
git config --global user.email kteng@vmware.com
git config --global user.name Kevin Teng
```

File: PowerValidatedSolutions.psm1 [hunk 3]
```diff
@@ -491,6 +491,81 @@ Function Set-vCenterPasswordExpiration {
...
+                                    Write-Error "Unable to find Cluster ($cluster) in vCenter Server ($vcfVcenterDetails.fqdn), check details and retry: PRE_VALIDATION_FAILED"
+                                }
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	Catch{
```
[the remark this answers is not in the document]
Comment (garlicNova) [reply PowerValidatedSolutions.psm1:563]: addressed in a87864654a54ba6dc3393bee232b77cf56a1b745

File: PowerValidatedSolutions.psm1 [hunk 4]
```diff
@@ -491,6 +491,81 @@ Function Set-vCenterPasswordExpiration {
...
+        - Validates that the workload domain exists in the SDDC Manager inventory
+        - Validates that network connectivity and authentication is possible to vCenter Server
+        - Gathers the ESXi hosts for the cluster specificed
+        - Configured all ESXi hosts in he provided cluster
+
+        EXAMPLE
+        Set-EsxiPasswordExpirationPolicy -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -cluster sfo-m01-cl01 -ExpirationInDays 60
+        This example configures all ESXi hosts within the cluster named sfo-m01-cl01 of the workload domain sfo-m01
+    #>
+	Param(
```
[the remark this answers is not in the document]
Comment (garlicNova) [reply PowerValidatedSolutions.psm1:512]: addressed in a87864654a54ba6dc3393bee232b77cf56a1b745

File: PowerValidatedSolutions.psm1 [hunk 5]
```diff
@@ -491,6 +491,81 @@ Function Set-vCenterPasswordExpiration {
...
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$pass,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$domain,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$cluster,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$ExpirationInDays,
+        [Parameter (Mandatory = $false)] [ValidateSet("true","false")] [String]$detail="true"
+	)
+	
+	Try{
```
[the remark this answers is not in the document]
Comment (garlicNova) [reply PowerValidatedSolutions.psm1:522]: addressed in a87864654a54ba6dc3393bee232b77cf56a1b745

File: PowerValidatedSolutions.psm1 (hunk 1 above)
Comment (garlicNova) [reply PowerValidatedSolutions.psm1:510]: addressed in 430834ddd05e1b61bda2cc01157674496e856c62

File: PowerValidatedSolutions.psm1 (hunk 2 above)
Comment (garlicNova) [reply PowerValidatedSolutions.psm1:546]: addressed in 430834ddd05e1b61bda2cc01157674496e856c62

**Review by tenthirtyam — APPROVED** [review]
All review comments addressed and resolved.

LGTM! 🚀

**Pull request merged**